django/db/models/constraints.py

```diff
@@ -69,14 +73,21 @@ def deconstruct(self):
         return path, args, kwargs
 
 
+DEFERRABLE_INITIALLY_DEFERRED = 'deferred'
+DEFERRABLE_INITIALLY_IMMEDIATE = 'immediate'
+
+
 class UniqueConstraint(BaseConstraint):
-    def __init__(self, *, fields, name, condition=None):
+    def __init__(self, *, fields, name, condition=None, defer=None):
```

Comment:
Not sure which of `defer` or `deferrable` makes more sense. The latter might be preferable if we choose the `Deferrable(Enum)` solution.

Reply:
I'm not sure either. Happy to paint the bikeshed either colour.

Reply:
On further thought, I'm leaning to `defer` (+0) because it's less typing.